scripts/add-validation-to-crds/parse-crds.go

```diff
@@ -223,6 +240,11 @@ func addValidationToRefs(props *apiextensions.JSONSchemaProps) error {
 		ruleYAML = refRuleWithKind
 	} else if signature == "external,kind,name,namespace" {
 		ruleYAML = refRuleWithKind
+		// kind is optional for projectRef (and maybe in future other well-known ref types)
+		// fieldPath is the best mechanism we have today (?)
```

Comment:
IIUC, when we migrate a resource to direct-based, we need to mark kind as "optional" for all its reference fields?

Reply:
It's not really about direct or not, it's about for project references, where we (briefly) introduced kind, should we continue to support the kind field?

I think if we drop kind, we would break anyone that added it in 1.118.1.  That should not be many people, but it is _some_ people.

`kind` optional and defaulting to Project (and only Project is allowed) shouldn't break anyone, though it is inelegant.